WordPress/src/main/java/org/wordpress/android/ui/people/PeopleListFragment.java

```diff
@@ -71,60 +78,173 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
 
         final ViewGroup rootView = (ViewGroup) inflater.inflate(R.layout.people_list_fragment, container, false);
 
-        mRecyclerView = (RecyclerView) rootView.findViewById(R.id.recycler_view);
-        mRecyclerView.setLayoutManager(new LinearLayoutManager(getActivity()));
-        mRecyclerView.setItemAnimator(new DefaultItemAnimator());
-        mRecyclerView.addItemDecoration(new PeopleItemDecoration(getActivity(), R.drawable.people_list_divider));
+        mLocalTableBlogID = getArguments().getInt(ARG_LOCAL_TABLE_BLOG_ID);
 
-        // progress bar that appears when loading more people
-        mProgress = (ProgressBar) rootView.findViewById(R.id.progress_footer);
-        mProgress.setVisibility(View.GONE);
+        mFilteredRecyclerView = (FilteredRecyclerView) rootView.findViewById(R.id.filtered_recycler_view);
+        mFilteredRecyclerView.addItemDecoration(new PeopleItemDecoration(getActivity(), R.drawable.people_list_divider));
+        mFilteredRecyclerView.setLogT(AppLog.T.PEOPLE);
+        mFilteredRecyclerView.setSwipeToRefreshEnabled(false);
 
-        return rootView;
-    }
+        mFilteredRecyclerView.setFilterListener(new FilteredRecyclerView.FilterListener() {
+            @Override
+            public List<FilterCriteria> onLoadFilterCriteriaOptions(boolean refresh) {
+                ArrayList<FilterCriteria> list = new ArrayList<>();
+                Collections.addAll(list, PeopleListFilter.values());
+                return list;
+            }
 
-    @Override
-    public void onActivityCreated(Bundle savedInstanceState) {
-        super.onActivityCreated(savedInstanceState);
+            @Override
+            public void onLoadFilterCriteriaOptionsAsync(FilteredRecyclerView.FilterCriteriaAsyncLoaderListener listener, boolean refresh) {
+                // no-op
+            }
 
-        mLocalTableBlogID = getArguments().getInt(ARG_LOCAL_TABLE_BLOG_ID);
+            @Override
+            public FilterCriteria onRecallSelection() {
+                mPeopleListFilter = AppPrefs.getPeopleListFilter();
+                return mPeopleListFilter;
+            }
 
-        // refresh the first page to serve fresh data
-        if (mOnFetchPeopleListener != null) {
-            mOnFetchPeopleListener.onFetchFirstPage();
-        }
+            @Override
+            public void onLoadData() {
+                updatePeople(false);
+            }
+
+            @Override
+            public void onFilterSelected(int position, FilterCriteria criteria) {
+                mPeopleListFilter = (PeopleListFilter) criteria;
+                AppPrefs.setPeopleListFilter(mPeopleListFilter);
+            }
+
+            @Override
+            public String onShowEmptyViewMessage(EmptyViewMessageType emptyViewMsgType) {
+                if (emptyViewMsgType == EmptyViewMessageType.NO_CONTENT) {
```

Comment:
minor thing here: you could have this case implemented inside the switch below, instead of having to make a distinction between `EmptyViewMessageType.NO_CONTENT` and the rest of the cases. This case should be handled by ` switch (emptyViewMsgType) `